### PR TITLE
apiextensions: promote CRD conversion to beta

### PR DESCRIPTION
**Pull request by sttts**
Requires https://github.com/kubernetes/kubernetes/pull/77743.

```release-note
CustomResourcesDefinition conversion via webhooks is promoted to beta. It requires that spec.preserveUnknownFields is set to false.
```

### CODE REVIEW

**Comment by jpbetz** [comment]
Looks correct to me. Do we typically include a test to ensure that the feature is disabled if the gate is turned off for beta features?

**Comment by liggitt** [comment]
shouldn't we also remove the `[Feature:CustomResourceWebhookConversion]` e2e tag so the e2e runs in non-alpha CI?

**Comment by sttts** [comment]
> shouldn't we also remove the `[Feature:CustomResourceWebhookConversion]` e2e tag so the e2e runs in non-alpha CI?

done

**Comment by liggitt** [comment]
e2e test failures are relevant:

```
[sig-api-machinery] CustomResourceConversionWebhook Should be able to convert a non homogeneous list of CRs expand_less	52s
test/e2e/apimachinery/crd_conversion_webhook.go:133
May 28 21:15:23.935: failed to create CustomResourceDefinition: 
CustomResourceDefinition.apiextensions.k8s.io "e2e-test-crd-webhook-3179-crds.stable.example.com" is invalid: spec.conversion.strategy: Invalid value: "Webhook": must be None if spec.preserveUnknownFields is true
test/utils/crd/crd_util.go:92

[sig-api-machinery] CustomResourceConversionWebhook Should be able to convert from CR v1 to CR v2 expand_less	35s
test/e2e/apimachinery/crd_conversion_webhook.go:110
May 28 21:26:15.237: failed to create CustomResourceDefinition: 
CustomResourceDefinition.apiextensions.k8s.io "e2e-test-crd-webhook-5605-crds.stable.example.com" is invalid: spec.conversion.strategy: Invalid value: "Webhook": must be None if spec.preserveUnknownFields is true
test/utils/crd/crd_util.go:92
```

**Comment by sttts** [comment]
> e2e test failures are relevant:

@liggitt conversion e2e test lacked the schema and `spec.preserveUnknownFields: fales`. Added that.

**Comment by sttts** [comment]
/retest

integation test without log file.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go [hunk 1]
```diff
@@ -343,7 +350,8 @@ func validateCustomResourceConversion(conversion *apiextensions.CustomResourceCo
 // ValidateCustomResourceDefinitionSpecUpdate statically validates
 func ValidateCustomResourceDefinitionSpecUpdate(spec, oldSpec *apiextensions.CustomResourceDefinitionSpec, established bool, fldPath *field.Path) field.ErrorList {
 	requireRecognizedVersion := oldSpec.Conversion == nil || hasValidConversionReviewVersionOrEmpty(oldSpec.Conversion.ConversionReviewVersions)
-	allErrs := validateCustomResourceDefinitionSpec(spec, requireRecognizedVersion, fldPath)
+
+	allErrs := validateCustomResourceDefinitionSpec(spec, requireRecognizedVersion, conversionAndPreserveUnknownFields(oldSpec), fldPath)
```
Comment (liggitt) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go]: webhook conversion was previously alpha, I don't think we have to do this check. let's follow-up to remove this and always require `preserveUnknownFields:false` when using webhook conversion.

**Comment by liggitt** [comment]
needs a rebase

**Comment by sttts** [comment]
Rebased.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go]: Follow-up https://github.com/kubernetes/kubernetes/pull/78501

**Comment by sttts** [comment]
Flake https://github.com/kubernetes/kubernetes/issues/72493

**Pull request merged**